### ISSUE TITLE
[Bug]: An error occurs when using full-text search in hybrid search, `vector type must be the same, field text_sparse_emb - type VECTOR_SPARSE_FLOAT, search info type VARCHAR`

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#36835]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:zhengbuqian-doc-in-restful-d174d05-20241010
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

When using text for full-text search, no error is reported. However, if full-text search is placed within a hybrid search, an error is reported, `inconsistent data types`.

```
[2024-10-14 11:47:24 - INFO - ci_test]: [initialize_milvus] Log cleaned up, start testing... (conftest.py:233)
[2024-10-14 11:47:24 - INFO - ci_test]: [setup_class] Start setup class... (client_base.py:40)
[2024-10-14 11:47:24 - INFO - ci_test]: *********************************** setup *********************************** (client_base.py:46)
[2024-10-14 11:47:24 - INFO - ci_test]: [setup_method] Start setup test case test_hybrid_search_with_full_text_search. (client_base.py:47)
-------------------------------- live log call ---------------------------------
[2024-10-14 11:47:24 - INFO - ci_test]: dataframe
        id       word                                           sentence                                          paragraph                                               text                                                emb
0        0     career                                      them at some.  stage us ok office sit rate. think cold marria...  election new risk along. start admit parent be...  [0.27244636354573304, 0.7408296153382901, 0.27...
1        1       same                       whose idea expect party far.  nothing water bank full close. drop strong fiv...  could popular world clearly lot. method star o...  [0.6846424443381437, 0.6248532417827315, 0.057...
2        2    mention  try offer citizen because discuss station arti...  three order rather network fund none. owner co...  month something their. all side focus once onl...  [0.646335454609506, 0.510131329932271, 0.52169...
3        3  necessary        discuss share month establish they account.  day financial red ahead watch design. notice r...  special moment fire loss best pick. mr full pl...  [0.30821210355810347, 0.6284092091728786, 0.39...
4        4       that                       account guess live continue.  worry page night design. discussion will road ...  field full include five middle goal. specific ...  [0.4793057757399589, 0.40508469740999853, 0.91...
...    ...        ...                                                ...                                                ...                                                ...                                                ...
4995  4995    mention  method finish show present of money everything...  none keep stage at him others herself enjoy. c...  say traditional view term. per admit ability e...  [0.4943923267718534, 0.17742261927500347, 0.05...
4996  4996  agreement                       continue probably per class.  season structure pull defense concern pay figu...  happen what guess and personal year three. fou...  [0.19608362344927366, 0.6988642892377759, 0.13...
4997  4997       game       mrs trial choice evening economy first drug.  word value nation past race have happen. toget...  force go along represent skin. meet threat fly...  [0.6653940038686026, 0.6788924970924798, 0.870...
4998  4998       read                    mean image western detail also.  agent night skill our boy. down real power ite...  themselves writer themselves list realize appr...  [0.8711416897477549, 0.988766712099613, 0.4073...
4999  4999    country  type conference become career value sense scor...  hundred matter tend ground anyone guy now baby...  pass adult effect school while benefit east he...  [0.42939873792678906, 0.5344311601453077, 0.23...

[5000 rows x 6 columns] (test_full_text_search.py:1803)
[2024-10-14 11:47:25 - INFO - ci_test]: Analyze document cost time: 0.06904387474060059 (common_func.py:338)
[2024-10-14 11:47:42 - ERROR - pymilvus.decorators]: RPC error: [hybrid_search], <MilvusException: (code=65535, message=fail to search on QueryNode 80: worker(79) query failed: Assert "static_cast<int>(field_meta.get_data_type()) == static_cast<int>(info.type())"  => vector type must be the same, field text_sparse_emb - type VECTOR_SPARSE_FLOAT, search info type VARCHAR at /go/src/github.com/milvus-io/milvus/internal/core/src/query/Plan.cpp:47
)>, <Time:{'RPC start': '2024-10-14 11:47:42.154218', 'RPC error': '2024-10-14 11:47:42.925732'}> (decorators.py:140)
[2024-10-14 11:47:42 - ERROR - ci_test]: Traceback (most recent call last):
  File "/Users/zilliz/workspace/milvus/tests/python_client/utils/api_request.py", line 32, in inner_wrapper
    res = func(*args, **_kwargs)
  File "/Users/zilliz/workspace/milvus/tests/python_client/utils/api_request.py", line 63, in api_request
    return func(*arg, **kwargs)
  File "/Users/zilliz/workspace/pymilvus/pymilvus/orm/collection.py", line 945, in hybrid_search
    resp = conn.hybrid_search(
  File "/Users/zilliz/workspace/pymilvus/pymilvus/decorators.py", line 141, in handler
    raise e from e
  File "/Users/zilliz/workspace/pymilvus/pymilvus/decorators.py", line 137, in handler
    return func(*args, **kwargs)
  File "/Users/zilliz/workspace/pymilvus/pymilvus/decorators.py", line 176, in handler
    return func(self, *args, **kwargs)
  File "/Users/zilliz/workspace/pymilvus/pymilvus/decorators.py", line 116, in handler
    raise e from e
  File "/Users/zilliz/workspace/pymilvus/pymilvus/decorators.py", line 86, in handler
    return func(*args, **kwargs)
  File "/Users/zilliz/workspace/pymilvus/pymilvus/client/grpc_handler.py", line 849, in hybrid_search
    return self._execute_hybrid_search(
  File "/Users/zilliz/workspace/pymilvus/pymilvus/client/grpc_handler.py", line 760, in _execute_hybrid_search
    raise e from e
  File "/Users/zilliz/workspace/pymilvus/pymilvus/client/grpc_handler.py", line 753, in _execute_hybrid_search
    check_status(response.status)
  File "/Users/zilliz/workspace/pymilvus/pymilvus/client/utils.py", line 63, in check_status
    raise MilvusException(status.code, status.reason, status.error_code)
pymilvus.exceptions.MilvusException: <MilvusException: (code=65535, message=fail to search on QueryNode 80: worker(79) query failed: Assert "static_cast<int>(field_meta.get_data_type()) == static_cast<int>(info.type())"  => vector type must be the same, field text_sparse_emb - type VECTOR_SPARSE_FLOAT, search info type VARCHAR at /go/src/github.com/milvus-io/milvus/internal/core/src/query/Plan.cpp:47
)>
 (api_request.py:45)
[2024-10-14 11:47:42 - ERROR - ci_test]: (api_response) : <MilvusException: (code=65535, message=fail to search on QueryNode 80: worker(79) query failed: Assert "static_cast<int>(field_meta.get_data_type()) == static_cast<int>(info.type())"  => vector type must be the same, field text_sparse_emb - type VECTOR_SPARSE_FLOAT, search info type VARCHAR at /go/s...... (api_request.py:46)
FAILED
testcases/test_full_text_search.py:1713 (TestHybridSearchWithFullTextSearch.test_hybrid_search_with_full_text_search[default-None-SPARSE_INVERTED_INDEX-True-True-0])
self = <test_full_text_search.TestHybridSearchWithFullTextSearch object at 0x13ac19760>
tokenizer = 'default', expr = None, enable_inverted_index = True
enable_partition_key = True, empty_percent = 0
index_type = 'SPARSE_INVERTED_INDEX'

    @pytest.mark.tags(CaseLabel.L0)
    @pytest.mark.parametrize("empty_percent", [0])
    @pytest.mark.parametrize("enable_partition_key", [True])
    @pytest.mark.parametrize("enable_inverted_index", [True])
    @pytest.mark.parametrize("index_type", ["SPARSE_INVERTED_INDEX"])
    @pytest.mark.parametrize("expr", [None, "text_match", "id_range"])
    @pytest.mark.parametrize("tokenizer", ["default"])
    def test_hybrid_search_with_full_text_search(
            self, tokenizer, expr, enable_inverted_index, enable_partition_key, empty_percent, index_type
    ):
        """
        target: test full text search
        method: 1. enable full text search and insert data with varchar
                2. search with text
                3. verify the result
        expected: full text search successfully and result is correct
        """
        tokenizer_params = {
            "tokenizer": tokenizer,
        }
        dim = 128
        fields = [
            FieldSchema(name="id", dtype=DataType.INT64, is_primary=True),
            FieldSchema(
                name="word",
                dtype=DataType.VARCHAR,
                max_length=65535,
                enable_tokenizer=True,
                tokenizer_params=tokenizer_params,
                is_partition_key=enable_partition_key,
            ),
            FieldSchema(
                name="sentence",
                dtype=DataType.VARCHAR,
                max_length=65535,
                enable_tokenizer=True,
                tokenizer_params=tokenizer_params,
            ),
            FieldSchema(
                name="paragraph",
                dtype=DataType.VARCHAR,
                max_length=65535,
                enable_tokenizer=True,
                tokenizer_params=tokenizer_params,
            ),
            FieldSchema(
                name="text",
                dtype=DataType.VARCHAR,
                max_length=65535,
                enable_tokenizer=True,
                enable_match=True,
                tokenizer_params=tokenizer_params,
            ),
            FieldSchema(name="emb", dtype=DataType.FLOAT_VECTOR, dim=dim),
            FieldSchema(name="text_sparse_emb", dtype=DataType.SPARSE_FLOAT_VECTOR),
        ]
        schema = CollectionSchema(fields=fields, description="test collection")
        bm25_function = Function(
            name="text_bm25_emb",
            function_type=FunctionType.BM25,
            input_field_names=["text"],
            output_field_names=["text_sparse_emb"],
            params={},
        )
        schema.add_function(bm25_function)
        data_size = 5000
        collection_w = self.init_collection_wrap(
            name=cf.gen_unique_str(prefix), schema=schema
        )
        fake = fake_en
        if tokenizer == "jieba":
            language = "zh"
            fake = fake_zh
        else:
            language = "en"
    
        data = [
            {
                "id": i,
                "word": fake.word().lower() if random.random() >= empty_percent else "",
                "sentence": fake.sentence().lower() if random.random() >= empty_percent else "",
                "paragraph": fake.paragraph().lower() if random.random() >= empty_percent else "",
                "text": fake.text().lower() if random.random() >= empty_percent else "",
                "emb": [random.random() for _ in range(dim)],
            }
            for i in range(data_size)
        ]
        df = pd.DataFrame(data)
        corpus = df["text"].to_list()
        log.info(f"dataframe\n{df}")
        texts = df["text"].to_list()
        word_freq = cf.analyze_documents(texts, language=language)
        tokens = list(word_freq.keys())
        if len(tokens) == 0:
            log.info(f"empty tokens, add a dummy token")
            tokens = ["dummy"]
        batch_size = 5000
        for i in range(0, len(df), batch_size):
            collection_w.insert(
                data[i : i + batch_size]
                if i + batch_size < len(df)
                else data[i : len(df)]
            )
            collection_w.flush()
        collection_w.create_index(
            "emb",
            {"index_type": "HNSW", "metric_type": "L2", "params": {"M": 16, "efConstruction": 500}},
        )
        collection_w.create_index(
            "text_sparse_emb",
            {
                "index_type": index_type,
                "metric_type": "BM25",
                "params": {
                    "bm25_k1": 1.5,
                    "bm25_b": 0.75,
                }
            }
        )
        if enable_inverted_index:
            collection_w.create_index("text", {"index_type": "INVERTED"})
        collection_w.load()
        nq=2
        sparse_search = AnnSearchRequest(
            data=[fake.text().lower() for _ in range(nq)],
            anns_field="text_sparse_emb",
            param={"metric_type": "IP"},
            limit=100,
        )
        dense_search = AnnSearchRequest(
            data=[[random.random() for _ in range(dim)] for _ in range(nq)],
            anns_field="emb",
            param={"metric_type": "L2"},
            limit=100,
    
        )
        # hybrid search
>       res, _ = collection_w.hybrid_search(
            reqs=[sparse_search,dense_search],
            rerank=WeightedRanker(0.5,0.5),
            limit=100,
            output_fields = ["id", "text"]
        )

test_full_text_search.py:1851: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../utils/wrapper.py:33: in inner_wrapper
    res, result = func(*args, **kwargs)
../base/collection_wrapper.py:188: in hybrid_search
    check_result = ResponseChecker(res, func_name, check_task, check_items, check,
../check/func_check.py:34: in run
    result = self.assert_succ(self.succ, True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <check.func_check.ResponseChecker object at 0x13ac2ad60>, actual = False
expect = True

    def assert_succ(self, actual, expect):
>       assert actual is expect, f"Response of API {self.func_name} expect {expect}, but got {actual}"
E       AssertionError: Response of API hybrid_search expect True, but got False
```

**Comment by zhuwenxing** [comment]
/assign @zhengbuqian 
PTAL

**Comment by zhuwenxing** [comment]
still reproduced in `milvus-io-master-aa904be-20241015`

```
[2024-10-15 14:39:26 - INFO - ci_test]: [initialize_milvus] Log cleaned up, start testing... (conftest.py:233)
[2024-10-15 14:39:26 - INFO - ci_test]: [setup_class] Start setup class... (client_base.py:40)
[2024-10-15 14:39:26 - INFO - ci_test]: *********************************** setup *********************************** (client_base.py:46)
[2024-10-15 14:39:26 - INFO - ci_test]: pymilvus version: 2.5.0rc96 (client_base.py:47)
[2024-10-15 14:39:26 - INFO - ci_test]: [setup_method] Start setup test case test_hybrid_search_with_full_text_search. (client_base.py:49)
-------------------------------- live log call ---------------------------------
[2024-10-15 14:39:26 - INFO - ci_test]: server version: aa904be-dev (client_base.py:165)
[2024-10-15 14:39:27 - INFO - ci_test]: dataframe
        id       word                                           sentence                                          paragraph                                               text                                                emb
0        0     career                                      them at some.  stage us ok office sit rate. think cold marria...  election new risk along. start admit parent be...  [0.008026716575471937, 0.5792498267987126, 0.1...
1        1       same                       whose idea expect party far.  nothing water bank full close. drop strong fiv...  could popular world clearly lot. method star o...  [0.10489974834779692, 0.6926362931189053, 0.04...
2        2    mention  try offer citizen because discuss station arti...  three order rather network fund none. owner co...  month something their. all side focus once onl...  [0.4247964376277872, 0.6315402459658459, 0.370...
3        3  necessary        discuss share month establish they account.  day financial red ahead watch design. notice r...  special moment fire loss best pick. mr full pl...  [0.34988937530546615, 0.22181978968433258, 0.0...
4        4       that                       account guess live continue.  worry page night design. discussion will road ...  field full include five middle goal. specific ...  [0.561396789119713, 0.7275136828019477, 0.8286...
...    ...        ...                                                ...                                                ...                                                ...                                                ...
4995  4995    mention  method finish show present of money everything...  none keep stage at him others herself enjoy. c...  say traditional view term. per admit ability e...  [0.3328382984862257, 0.9633145731913216, 0.936...
4996  4996  agreement                       continue probably per class.  season structure pull defense concern pay figu...  happen what guess and personal year three. fou...  [0.1644241327160335, 0.1119050634821741, 0.832...
4997  4997       game       mrs trial choice evening economy first drug.  word value nation past race have happen. toget...  force go along represent skin. meet threat fly...  [0.8225183928326887, 0.08922623563693477, 0.49...
4998  4998       read                    mean image western detail also.  agent night skill our boy. down real power ite...  themselves writer themselves list realize appr...  [0.6215162399375951, 0.6217949594931688, 0.653...
4999  4999    country  type conference become career value sense scor...  hundred matter tend ground anyone guy now baby...  pass adult effect school while benefit east he...  [0.5179315873195036, 0.37077905542400247, 0.97...

[5000 rows x 6 columns] (test_full_text_search.py:1990)
[2024-10-15 14:39:27 - INFO - ci_test]: Analyze document cost time: 0.06556987762451172 (common_func.py:340)
[2024-10-15 14:39:44 - ERROR - pymilvus.decorators]: RPC error: [hybrid_search], <MilvusException: (code=65535, message=fail to search on QueryNode 16: worker(16) query failed: Assert "static_cast<int>(field_meta.get_data_type()) == static_cast<int>(info.type())"  => vector type must be the same, field text_sparse_emb - type VECTOR_SPARSE_FLOAT, search info type VARCHAR at /go/src/github.com/milvus-io/milvus/internal/core/src/query/Plan.cpp:47
)>, <Time:{'RPC start': '2024-10-15 14:39:43.867999', 'RPC error': '2024-10-15 14:39:44.926685'}> (decorators.py:140)
[2024-10-15 14:39:44 - ERROR - ci_test]: Traceback (most recent call last):
  File "/Users/zilliz/workspace/milvus/tests/python_client/utils/api_request.py", line 32, in inner_wrapper
    res = func(*args, **_kwargs)
  File "/Users/zilliz/workspace/milvus/tests/python_client/utils/api_request.py", line 63, in api_request
    return func(*arg, **kwargs)
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 945, in hybrid_search
    resp = conn.hybrid_search(
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/decorators.py", line 141, in handler
    raise e from e
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/decorators.py", line 137, in handler
    return func(*args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/decorators.py", line 176, in handler
    return func(self, *args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/decorators.py", line 116, in handler
    raise e from e
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/decorators.py", line 86, in handler
    return func(*args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 854, in hybrid_search
    return self._execute_hybrid_search(
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 765, in _execute_hybrid_search
    raise e from e
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 758, in _execute_hybrid_search
    check_status(response.status)
  File "/Users/zilliz/opt/anaconda3/envs/full_text_search/lib/python3.8/site-packages/pymilvus/client/utils.py", line 63, in check_status
    raise MilvusException(status.code, status.reason, status.error_code)
pymilvus.exceptions.MilvusException: <MilvusException: (code=65535, message=fail to search on QueryNode 16: worker(16) query failed: Assert "static_cast<int>(field_meta.get_data_type()) == static_cast<int>(info.type())"  => vector type must be the same, field text_sparse_emb - type VECTOR_SPARSE_FLOAT, search info type VARCHAR at /go/src/github.com/milvus-io/milvus/internal/core/src/query/Plan.cpp:47
)>
 (api_request.py:45)
[2024-10-15 14:39:44 - ERROR - ci_test]: (api_response) : <MilvusException: (code=65535, message=fail to search on QueryNode 16: worker(16) query failed: Assert "static_cast<int>(field_meta.get_data_type()) == static_cast<int>(info.type())"  => vector type must be the same, field text_sparse_emb - type VECTOR_SPARSE_FLOAT, search info type VARCHAR at /go/s...... (api_request.py:46)
FAILED
```

**Comment by zhengbuqian** [comment]
this is caused by providing IP metric when searching. 

the error message is not straightforward though, fixing in https://github.com/milvus-io/milvus/pull/36910

**Comment by zhengbuqian** [comment]
/assign @zhuwenxing
/unassign
https://github.com/milvus-io/milvus/pull/36910 has been merged, please verify

**Comment by zhuwenxing** [comment]
verified and fixed in `master-20750c0-20241017`
If the parameters are correct, the hybrid search will be successful. If the parameters are incorrect, the error message will also meet expectations.
